patches/tModLoader/Terraria/ModLoader/ChestLoot.cs

```diff
@@ -0,0 +1,321 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Terraria.Enums;
+using Terraria.GameContent.Events;
+using Terraria.ID;
+using Terraria.Localization;
+
+namespace Terraria.ModLoader;
+
+// TODO: AddPylon method at some point?
+public readonly struct ChestLoot {
+	public interface ICondition {
+		string Description { get; }
+
+		bool IsAvailable();
+	}
+
+	public sealed class Condition : ICondition {
+		#region Common conditions
+		public static readonly Condition Christmas = new(NetworkText.Empty, () => Main.xMas);
+		public static readonly Condition Halloween = new(NetworkText.Empty, () => Main.halloween);
+		public static readonly Condition Hardmode = new(NetworkText.Empty, () => Main.hardMode);
+		public static readonly Condition PreHardmode = new(NetworkText.Empty, () => !Main.hardMode);
+		public static readonly Condition TimeDay = new(NetworkText.Empty, () => Main.dayTime);
+		public static readonly Condition TimeNight = new(NetworkText.Empty, () => !Main.dayTime);
+		public static readonly Condition NotBloodMoon = new(NetworkText.Empty, () => !Main.bloodMoon);
+		public static readonly Condition BloodMoon = new(NetworkText.Empty, () => Main.bloodMoon);
+		public static readonly Condition Eclipse = new(NetworkText.Empty, () => Main.eclipse);
+		public static readonly Condition NotEclipse = new(NetworkText.Empty, () => !Main.eclipse);
+		public static readonly Condition RemixWorld = new(NetworkText.Empty, () => Main.remixWorld);
+		public static readonly Condition TenthAnniversary = new(NetworkText.Empty, () => Main.tenthAnniversaryWorld);
+		public static readonly Condition BirthdayPartyIsUp = new(NetworkText.Empty, () => BirthdayParty.PartyIsUp);
+		public static readonly Condition NightLanternsUp = new(NetworkText.Empty, () => LanternNight.LanternsUp);
+		public static readonly Condition IsMoonFull = new(NetworkText.Empty, () => Main.GetMoonPhase() == MoonPhase.Full);
+		public static readonly Condition IsMoonWaningGibbous = new(NetworkText.Empty, () => Main.GetMoonPhase() == MoonPhase.ThreeQuartersAtLeft);
+		public static readonly Condition IsMoonThirdQuarter = new(NetworkText.Empty, () => Main.GetMoonPhase() == MoonPhase.HalfAtLeft);
+		public static readonly Condition IsMoonWaningCrescent = new(NetworkText.Empty, () => Main.GetMoonPhase() == MoonPhase.QuarterAtLeft);
+		public static readonly Condition IsMoonNew = new(NetworkText.Empty, () => Main.GetMoonPhase() == MoonPhase.Empty);
+		public static readonly Condition IsMoonWaxingCrescent = new(NetworkText.Empty, () => Main.GetMoonPhase() == MoonPhase.QuarterAtRight);
```

Comment:
We will want to replace the NetworkText.Empty with something meaningful before merge.